storage/src/test/scala/ch/epfl/bluebrain/nexus/storage/routes/AuthDirectivesSpec.scala

```diff
@@ -4,98 +4,111 @@ import akka.http.scaladsl.model.StatusCodes
+          None,
+          None,
+          false,
+          Some(subject)
+        )
+        .value
+
+    implicit val anonymousMethod: AuthorizationMethod =
+      VerifyToken(validIssuer, validSubject, None, new JWKSet(rsaKey.toPublicJWK))
+
+    "Succeed with a valid token" in {
+      val token = generateToken(validSubject, validIssuer, rsaKey)
+      Get("/").addCredentials(OAuth2BearerToken(token)) ~> validateRoute ~> check {
         status shouldEqual StatusCodes.OK
       }
     }
 
-    "fail the route" when {
+    "Fail with an invalid issuer" in {
+      val token = generateToken(validSubject, "xxx", rsaKey)
+      Get("/").addCredentials(OAuth2BearerToken(token)) ~> validateRoute ~> check {
+        status shouldEqual StatusCodes.Unauthorized
+      }
+    }
 
-      "the client throws an error for caller" in {
-        implicit val token: Option[AccessToken] = None
-        deltaIdentities()(any[Option[AccessToken]]) shouldReturn
-          Task.raiseError(DeltaIdentitiesClientError.IdentitiesServerStatusError(StatusCodes.InternalServerError, ""))
-        val route                               = Routes.wrap(extractCaller.apply(_ => complete("")))
-        Get("/") ~> route ~> check {
-          status shouldEqual StatusCodes.InternalServerError
-        }
+    "Fail with an invalid subject" in {
+      val token = generateToken("bob", validIssuer, rsaKey)
+      Get("/").addCredentials(OAuth2BearerToken(token)) ~> validateRoute ~> check {
+        status shouldEqual StatusCodes.Unauthorized
       }
-      "the client returns Unauthorized for caller" in {
-        implicit val token: Option[AccessToken] = None
-        deltaIdentities()(any[Option[AccessToken]]) shouldReturn
-          Task.raiseError(DeltaIdentitiesClientError.IdentitiesClientStatusError(StatusCodes.Unauthorized, ""))
-        val route                               = Routes.wrap(extractCaller.apply(_ => complete("")))
-        Get("/") ~> route ~> check {
-          status shouldEqual StatusCodes.Unauthorized
-        }
+    }
+
+    "Fail with a token signed with another key" in {
+      val anotherKey: RSAKey = new RSAKeyGenerator(2048).keyID(genString()).generate()
+      val token              = generateToken(validSubject, validIssuer, anotherKey)
+      Get("/").addCredentials(OAuth2BearerToken(token)) ~> validateRoute ~> check {
+        status shouldEqual StatusCodes.Unauthorized
       }
-      "the client returns Forbidden for caller" in {
-        implicit val token: Option[AccessToken] = None
-        deltaIdentities()(any[Option[AccessToken]]) shouldReturn
-          Task.raiseError(DeltaIdentitiesClientError.IdentitiesClientStatusError(StatusCodes.Forbidden, ""))
-        val route                               = Routes.wrap(extractCaller.apply(_ => complete("")))
-        Get("/") ~> route ~> check {
-          status shouldEqual StatusCodes.Forbidden
-        }
+    }
+
+    "Fail with an invalid token" in {
```

Comment:
Worth checking expired and not yet active tokens?

Reply:
This is already tested elsewhere and handled when parsing the token, I did not want to cover all the cases again here